### DISCUSSION TITLE
restAPI file upload

### DISCUSSION THREAD

**Discussion by LWJ-SARA** [netbox-community/netbox#17272] [Q&A]
When I was testing with postman, I wanted to upload a csv file with a post request to add the IP address, but when I sent the request, it always returned {
"address": [
"This field is required."
]
}
In fact, I already have this field in my file, and I try to upload a txt file that contains json data, and the same problem occurs.
I think I need a detailed guide. Thank you so much~

**Comment by candlerb** [comment]
You haven't included *any* information which could be used to help you or to reproduce your problem: in particular you haven't said the URL path that you tried to upload to, or the body of your upload, or even the version of Netbox you're running.

https://www.catb.org/~esr/faqs/smart-questions.html#intro

**Reply by LWJ-SARA** [reply]
URL:http://172.16.214.130:8000/api/ipam/ip-addresses/
method:POST
netbox version:v4.0.9
[netbox_IP addresses.csv](https://github.com/user-attachments/files/16758653/netbox_IP.addresses.csv)

**Reply by candlerb** [reply]
I don't think it says anywhere that you can send a CSV file to this REST API endpoint - if you've found somewhere that does, please let me know where. The REST API is documented [here](https://netboxlabs.com/docs/netbox/en/stable/integrations/rest-api/).

There is however a bulk creation endpoint where you send a JSON list of objects. That's documented [here](https://netboxlabs.com/docs/netbox/en/stable/integrations/rest-api/#creating-multiple-objects).

**Reply by LWJ-SARA** [reply]
Ok, thank you! Then I will try to parse the csv file into JSON format for operation.
